hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/ozone/OzoneClientUtils.java

```diff
@@ -179,21 +190,52 @@ public static InetSocketAddress getScmDataNodeBindAddress(
       Configuration conf, String ... keys) {
     for (final String key : keys) {
       final String value = conf.getTrimmed(key);
-      if (value != null && !value.isEmpty()) {
-        String[] splits = value.split(":");
-
-        if(splits.length < 1 || splits.length > 2) {
-          throw new IllegalArgumentException(
-              "Invalid value " + value + " for config key " + key +
-                  ". It should be in 'host' or 'host:port' format");
-        }
-        return Optional.of(splits[0]);
+      Optional<String> splits = getHostName(value);
+      if (splits.isPresent()) {
+        return splits;
       }
     }
     return Optional.absent();
   }
 
   /**
+   * Returns a hostname from the hostname:port or hostname.
+   * @param value
+   * @return
```

Comment:
fixed. Thanks for the comment.  I have replaced the parsing code with HostAndPort class. So this code has becomes simpler and I have updated the Javadocs. So we don't need the pos parameter any longer.